arangod/VocBase/Methods/Collections.cpp

```diff
@@ -263,14 +264,38 @@ Result Collections::create(TRI_vocbase_t& vocbase,
                info.collectionType != TRI_col_type_e::TRI_COL_TYPE_EDGE) {
       return TRI_ERROR_ARANGO_COLLECTION_TYPE_INVALID;
     }
+
     TRI_ASSERT(info.properties.isObject());
     helper.clear();
     helper.openObject();
     helper.add(arangodb::StaticStrings::DataSourceType,
                arangodb::velocypack::Value(static_cast<int>(info.collectionType)));
     helper.add(arangodb::StaticStrings::DataSourceName,
                arangodb::velocypack::Value(info.name));
+
+
+    auto replicationFactorSlice = info.properties.get(StaticStrings::ReplicationFactor);
+    if(replicationFactorSlice.isNone()) {
+      auto factor = vocbase.replicationFactor();
+      if(factor > 0 && vocbase.IsSystemName(info.name)) {
+        auto* cl = application_features::ApplicationServer::lookupFeature<ClusterFeature>("Cluster");
+        factor = std::max(vocbase.replicationFactor(), cl->systemReplicationFactor());
+      }
+      helper.add(StaticStrings::ReplicationFactor, VPackValue(factor));
+    }
+
+    // system collections will be sharded normally - we avoid a self reference when creating _graphs
+    if(vocbase.sharding() == "single" && !vocbase.IsSystemName(info.name)) {
+      auto distribute = info.properties.get(StaticStrings::DistributeShardsLike);
+      if(distribute.isNone()) {
+        helper.add(StaticStrings::DistributeShardsLike, VPackValue(StaticStrings::GraphCollection));
+      } else if (distribute.isString() && distribute.compareString("")) {
+        helper.add(StaticStrings::DistributeShardsLike, VPackSlice::nullSlice()); //delete empty string from info slice
```

Comment:
doesn't this also overwrite the user's `distributeShardsLike` choice, e.g. if the user specified `{ distributeShardsLike: "obi" }`?

Reply:
= 0 missing